core/slice.go

```diff
@@ -455,37 +458,16 @@ func (sl *Slice) pcrc(batch ethdb.Batch, header *types.Header, domTerminus commo
 	return termini[location.SubIndex()], newTermini, nil
 }
 
-// HLCR Hierarchical Longest Chain Rule compares externTd to the currentHead Td and returns true if externTd is greater
-func (sl *Slice) hlcr(externTd *big.Int) bool {
-	currentTd := sl.hc.GetTdByHash(sl.hc.CurrentHeader().Hash())
-	log.Debug("HLCR:", "Header hash:", sl.hc.CurrentHeader().Hash(), "currentTd:", currentTd, "externTd:", externTd)
-	reorg := currentTd.Cmp(externTd) < 0
-	//TODO need to handle the equal td case
-	// https://github.com/dominant-strategies/go-quai/issues/430
+// POEM compares externS to the currentHead S and returns true if externS is greater
+func (sl *Slice) poem(externS *big.Int, currentS *big.Int) bool {
+	log.Debug("POEM:", "Header hash:", sl.hc.CurrentHeader().Hash(), "currentS:", common.BigBitsToBits(currentS), "externS:", common.BigBitsToBits(externS))
+	reorg := currentS.Cmp(externS) < 0
 	return reorg
 }
 
-// CalcTd calculates the TD of the given header using PCRC.
-func (sl *Slice) calcTd(header *types.Header, domTd *big.Int, domOrigin bool) (*big.Int, error) {
-	priorTd := sl.hc.GetTd(header.ParentHash(), header.NumberU64()-1)
-	if priorTd == nil {
-		return nil, consensus.ErrFutureBlock
-	}
-
-	Td := priorTd.Add(priorTd, header.Difficulty())
-
-	if domOrigin {
-		// If its a dom block we don't compute the td, instead just return the
-		// td given by dom
-		return domTd, nil
-	}
-
-	return Td, nil
-}
-
 // GetPendingHeader is used by the miner to request the current pending header
 func (sl *Slice) GetPendingHeader() (*types.Header, error) {
-	if ph := sl.phCache[sl.bestPhKey].Header; ph != nil {
+	if ph := sl.phCache[sl.bestPhKey.Key()].Header; ph != nil {
```

Comment:
(nit) the naming here is confusing. I would think `bestPhKey` is the key, but then you call `bestPhKey.Key()` to convert it into a key. So I guess `bestPhKey` isn't actually a key?

Reply:
can you suggest a better name please

Reply:
make it a separate file. If you have a better name please suggest.

Reply:
Calling it bestPhKeyObject now in true OOP style

Reply:
> Calling it bestPhKeyObject now in true OOP style

:roll_eyes: 

> If you have a better name please suggest.

Well it looks to me like the object isn't a key. It is a collection of several things, only one of which is a key...

```
type BestPhKeyObject struct {
	key       common.Hash
	entropy   *big.Int
	blockHash common.Hash
}
```
Is there some logic for why these three seemingly unrelated data need to be grouped into one object? That logic should tell you your name. Conversely, if there is no logical reason to group these together... why are they grouped together?